tidy3d/components/data/monitor_data.py

```diff
@@ -133,7 +133,136 @@ def symmetry_expanded_copy(self) -> AbstractFieldData:
+
+        # Interpolate tangential field components to cell centers
+        interp_dict = dict(zip(tan_dims, centers))
+        centered_fields = {key: val.interp(**interp_dict) for key, val in tan_fields.items()}
+        return centered_fields
+
+    @property
+    def poynting(self) -> xr.DataArray:
+        """Time-averaged Poynting vector for frequency-domain data associated to a 2D monitor."""
+
+        # Tangential fields are ordered as E1, E2, H1, H2
+        tan_fields = self._centered_tangential_fields
+        field_list = list(tan_fields.values())
+        e_x_h_star = field_list[0] * field_list[3].conj() - field_list[1] * field_list[2].conj()
+        poynting = 0.5 * np.real(e_x_h_star)
+        # cast to xr.DataArray as this is technically no longer a ScalarFieldArray
+        return xr.DataArray(poynting)
+
+    @property
+    def flux(self) -> xr.DataArray:
+        """Flux for data corresponding to a 2D monitor."""
+
+        # Compute flux by integrating Poynting vector in-plane
+        d_area = self._diff_area
+        return (self.poynting * d_area).sum(dim=d_area.dims)
+
+    def dot(
+        self, field_data: Union[FieldData, ModeSolverData], conjugate: bool = True
+    ) -> xr.DataArray:
+        """Dot product (modal overlap) with another ``field_data`` object. Both datasets have to be
+        frequency-domain data associated with a 2D monitor. Along the tangential directions,
+        the datasets have to have the same discretization. Along the normal direction, the monitor
+        position may differ and is ignored. Other coordinates (``frequency``, ``mode_index``) have
+        to be either identical or broadcastable.
+
+        Parameters
+        ----------
+        field_data : :class:`ElectromagneticFieldData`
+            A data instance to compute the dot product with.
+        conjugate : bool, optional
+            If ``True`` (default), the dot product is defined as ``1 / 4`` times the integral of
+            ``E1* x H2 - H1* x E2``, where ``x`` is the cross product, ``*`` is complex conjugation,
+            and ``(E1, H1)``, ``(E2, H2)`` are the two sets of field data. If ``False``, the
+            complex conjugation is skipped.
+
+        Note
+        ----
+            The dot product with and without conjugation is equivalent (up to a phase) for
+            modes in lossless waveguides but differs for modes in lossy materials. In that case,
+            the conjugated dot product can be interpreted as the fraction of the power of the first
+            mode carried by the second, but modes are not orthogonal with respect to that product
+            and the sum of carried power fractions exceed 1. In the non-conjugated definition,
+            orthogonal modes can be defined, but the interpretation of modal overlap as power
+            carried by a given mode is no longer valid.
+        """
+
+        # Tangential fields for current and input field data
```

Comment:
Like before, I think we should just do the following lines with the dictionary keys (or comprehension) instead of assuming an ordering.

```py
       field_cmps_self = self._centered_tangential_fields
       # pylint:disable=protected-access
       field_cmps_other = field_data._centered_tangential_fields
       if conjugate:
           field_cmps_self = {name: field.conj() for name: field in field_cmps_self.items()}

         # Cross products of fields
         e_self_x_h_other = field_cmps_self['E1'] * field_cmps_other['H2'] - field_cmps_self['E2'] * field_cmps_other['H1']
         h_self_x_e_other = field_cmps_self['H2'] * field_cmps_other['H1'] - field_cmps_self['E1'] * field_cmps_other['H2']

         # Integrate over plane
         d_area = self._diff_area
         return 0.25 * xr.DataArray((e_self_x_h_other - h_self_x_e_other) * d_area).sum(dim=d_area.dims)


```